openhcl/underhill_core/src/nvme_manager.rs

```diff
@@ -271,20 +325,344 @@ impl VfioNvmeDriverSpawner {
         dma_client: Arc<dyn user_driver::DmaClient>,
     ) -> Result<nvme_driver::NvmeDriver<VfioDevice>, NvmeSpawnerError> {
         let device = VfioDevice::new(driver_source, pci_id, dma_client)
-            .instrument(tracing::info_span!("vfio_device_open", pci_id))
+            .instrument(tracing::info_span!("nvme_vfio_device_open", pci_id))
             .await
             .map_err(NvmeSpawnerError::Vfio)?;
 
         // TODO: For now, any isolation means use bounce buffering. This
         // needs to change when we have nvme devices that support DMA to
         // confidential memory.
         nvme_driver::NvmeDriver::new(driver_source, vp_count, device, is_isolated)
-            .instrument(tracing::info_span!("nvme_driver_init", pci_id))
+            .instrument(tracing::info_span!("nvme_driver_new", pci_id))
             .await
             .map_err(NvmeSpawnerError::DeviceInitFailed)
     }
 }
 
+mod device_manager {
```

Comment:
i'm not really a fan of inline modules once they get larger than some definitions - can we move this to its own file?

Reply:
Happy to

Reply:
I pushed a first draft of this split before I ran out of time. Let me know what you think.